redis/cache.py

```diff
@@ -0,0 +1,95 @@
+import random
+import time
+from collections import OrderedDict, defaultdict
+
+
+class _Cache:
+    def __init__(self, max_size: int, ttl: int, eviction_policy: str, **kwargs):
+        self.max_size = max_size
+        self.ttl = ttl
+        self.eviction_policy = eviction_policy
```

Comment:
We should have a list of policies mapped in this class. Then, map them to [enum](https://docs.python.org/3/library/enum.html) to make to make it available during configuration